lib/api/core/auth/tokenManager.js

```diff
@@ -172,22 +169,20 @@ class TokenManager {
       pos = this.tokens.search({idx: oldIndex});
 
     // If the old token has been created and then refreshed within the same
-    // second, then it has the exact same caracteristics than the new one.
+    // second, then it has the exact same characteristics than the new one.
     // This should never happen, though, especially if we add at least 1
     // real-time subscribe in the middle of the login+refresh sequence (all
     // within the same second) but, oh, well... it costs nothing to fix a
     // potentially very, very, very hard to debug random problem before it
     // occurs
     if (pos > -1 && oldToken._id !== newToken._id) {
-      this.tokens.insert({
-        idx: getTokenIndex(newToken),
-        connectionId: this.tokens.array[pos].connectionId,
-        _id: newToken._id,
-        expiresAt: newToken.expiresAt,
-        userId: newToken.userId,
-        rooms: this.tokens.array[pos].rooms
-      });
-      this.tokens.remove({idx: oldIndex});
+      this._add(
+        newToken,
+        this.tokens.array[pos].connectionId,
+        [...this.tokens.array[pos].rooms]
+      );
```

Comment:
(nit)
this should be on the line above according to the new wrapping rules

Reply:
:+1: